### DISCUSSION TITLE
Drawn objects get saved multiple times

### DISCUSSION THREAD

**Discussion by SimpsonsWorld** [geoman-io/leaflet-geoman#1342] [Questions & Help] [answered]
I tried to make a save-function to save the drawn items in a geojson. It just works but it always stores the items multiple times. 
This is my function:
```
function saveToDb() {
  localStorage.clear();
  var allLayers = new L.featureGroup();
  map.eachLayer(function (layer) {
    if (layer instanceof L.Path) {
      allLayers.addLayer(layer);
    } else {
      if (layer instanceof L.Marker) {
        allLayers.addLayer(layer);
      }
    }
  });
  var geojson = allLayers.toGeoJSON();
  localStorage.setItem("data", JSON.stringify(geojson));
}
```

I would be very thankful if you could help me with this problem.

**Comment by Falke-Design** [comment]
The code looks good. Maybe the problem is while loading the layers from the LocalStorage? 
Also I suggest you to use [`removeItem`](https://developer.mozilla.org/en-US/docs/Web/API/Storage/removeItem) instead of `localStorage.clear()`.

To make it a little bit easier you can use `map.pm.getGeomanLayers(true)` to get all the Layers on the map, which can be modified by Geoman:

```
localStorage.removeItem("data");
var geojson = map.pm.getGeomanLayers(true).toGeoJSON();
localStorage.setItem("data", JSON.stringify(geojson));
```

**Reply by SimpsonsWorld** [reply]
I just found out that just after clicking on the save button the Local Storage looks quite good - every Item is only saved once. But when I am refreshing the site, the multiple items show up. Any idea where this problem can be located?

**Reply by Falke-Design** [reply]
Pretty hard to say without the code

**Reply by SimpsonsWorld** [reply]
I'm sorry, I am new here. So I hope it's okay if I share my could like this.
[Karte.txt](https://github.com/geoman-io/leaflet-geoman/files/11136367/Karte.txt)

**Reply by Falke-Design** [reply, marked as answer]
You are executing this 4 times:
```
 var data = localStorage.getItem('data');
      if(data){
         var geoLayer = L.geoJSON(JSON.parse(data)).addTo(map);
      } 
```
This added the same layers multiple times

**Reply by SimpsonsWorld** [reply]
Thank you so much! Don't know why I haven't seen it.
Can you also help me with the saving of the properties (like color, text,...). I saw many had the same problem here that the properties don't get saved but I didn't really found a solution.
